modules/core/src/main/java/org/apache/synapse/deployers/LocalEntryDeployer.java

```diff
@@ -79,6 +102,66 @@ public String deploySynapseArtifact(OMElement artifactConfig, String fileName,
         return null;
     }
 
+    private void handleHttpConnectorCertificates(OMElement element) throws DeploymentException {
+
+        OMElement httpInitElement =
+                element.getFirstChildWithName(new QName(SynapseConstants.SYNAPSE_NAMESPACE, HTTP_CONNECTION_IDENTIFIER));
+        if (httpInitElement != null) {
+            Iterator childElementIterator = httpInitElement.getChildElements();
+            while (childElementIterator.hasNext()) {
+                OMElement childElement = (OMElement) childElementIterator.next();
+                String childElementValue = childElement.getText();
+                String transformedElementValue = getTransformedElementValue(childElementValue);
+                if (transformedElementValue.endsWith(CERTIFICATE_EXTENSION)) {
+                    loadCertificateFileToStore(transformedElementValue);
+                }
+            }
+        }
+    }
+
+    private void loadCertificateFileToStore(String certificateFileResourceKey) throws DeploymentException {
+
+        String certificateFilePath = getSynapseConfiguration().getRegistry().getRegistryEntry(certificateFileResourceKey).getName();
+        File certificateFile = new File(certificateFilePath);
+        String certificateAlias = certificateFile.getName().split("\\.")[0];
+        try {
+            FileInputStream certificateFileInputStream = FileUtils.openInputStream(new File(certificateFilePath));
+            SslSenderTrustStoreHolder sslSenderTrustStoreHolder = SslSenderTrustStoreHolder.getInstance();
+            KeyStore sslSenderTrustStore = sslSenderTrustStoreHolder.getKeyStore();
+            CertificateFactory certificateFactory = CertificateFactory.getInstance("X.509");
+            Certificate certificate = certificateFactory.generateCertificate(certificateFileInputStream);
+            sslSenderTrustStore.setCertificateEntry(certificateAlias, certificate);
+
+            FileOutputStream fileOutputStream = new FileOutputStream(sslSenderTrustStoreHolder.getLocation());
+            sslSenderTrustStore.store(fileOutputStream, sslSenderTrustStoreHolder.getPassword().toCharArray());
+
+            FileInputStream fileInputStream = new FileInputStream(sslSenderTrustStoreHolder.getLocation());
+            InputStream dest = IOUtils.toBufferedInputStream(fileInputStream);
+            fileInputStream.close();
+            sslSenderTrustStore.load(dest, sslSenderTrustStoreHolder.getPassword().toCharArray());
+            dest.close();
+
+            sslSenderTrustStoreHolder.setKeyStore(sslSenderTrustStore);
+            KeyStoreReloaderHolder.getInstance().reloadAllKeyStores();
+        } catch (CertificateException | IOException | KeyStoreException | NoSuchAlgorithmException e) {
+            throw new RuntimeException(e);
```

Comment:
Rather than throwing a generic runtime exception, shall we add a more descriptive message?

ex: ("Failed to load certificate file to trust store", e)